pkg/graph/resolvers/mto_milestone_test.go

```diff
@@ -516,3 +522,65 @@ func (suite *ResolverSuite) TestMTOMilestoneUpdateLinkedSolutions_UnlinkByCommon
 	suite.NoError(err)
 	suite.Len(linkedSolutions, 0)
 }
+
+func (suite *ResolverSuite) TestMTOMilestoneNoLinkedSolutions_SinglePlan() {
+	plan := suite.createModelPlan("NoLinkedSolutions SinglePlan")
+	_ = suite.createMTOSolutionCommon(plan.ID, models.MTOCSKInnovation, nil)
+
+	linkedMilestone := suite.createMilestoneCommon(
+		plan.ID,
+		models.MTOCommonMilestoneKeyManageCd,
+		[]models.MTOCommonSolutionKey{models.MTOCSKInnovation},
+	)
+
+	unlinkedMilestone := suite.createMilestoneCommon(
+		plan.ID,
+		models.MTOCommonMilestoneKeyRevColBids,
+		nil,
+	)
+
+	nonLinkedMilestones := suite.getMilestonesWithNoLinkedSolution(plan.ID)
+	milestoneIDs := lo.Map(nonLinkedMilestones, func(m *models.MTOMilestone, _ int) uuid.UUID {
+		return m.ID
+	})
+
+	suite.Contains(milestoneIDs, unlinkedMilestone.ID, "Unlinked milestone should be present")
+	suite.NotContains(milestoneIDs, linkedMilestone.ID, "Linked milestone should be excluded from the results")
+}
+
+func (suite *ResolverSuite) TestMTOMilestoneNoLinkedSolutions_MultiplePlans() {
```

Comment:
@OddTomBrooks , I'm not sure that this is actually verifying the loader.  If you take a look at `TestMTOMilestoneGetBySolutionIDLoader`, the helper function takes a list of keys, and expected results. It then uses go routines to ensure that test cases hit the data loader at the same time, could you  try to use that helper method here please?